lib/WorseReflection/Bridge/Phpactor/DocblockParser/PHPStanParsedDocblock.php

```diff
@@ -0,0 +1,321 @@
+            }
+            $this->convertType($methodTag->returnType);
+        }
+
+        return TypeFactory::undefined();
+    }
+
+    public function inherits(): bool
+    {
+        return false;
+    }
+
+    public function vars(): DocBlockVars
+    {
+        $vars = [];
+        foreach ($this->node->getVarTagValues() as $varTag) {
+            assert($varTag instanceof VarTagValueNode);
+            $vars[] = new DocBlockVar(
+                ltrim($varTag->variableName, '$'),
+                $this->convertType($varTag->type),
+            );
+        }
+
+        return new DocBlockVars($vars);
+    }
+
+    public function params(): DocBlockParams
+    {
+        $params = [];
+        foreach ($this->node->getParamTagValues() as $paramTag) {
+            $params[] = new DocBlockParam(
+                ltrim($paramTag->parameterName, '$'),
+                $this->convertType($paramTag->type),
+            );
+        }
+
+        return new DocBlockParams($params);
+    }
+
+    public function parameterType(string $paramName): Type
+    {
+        $types = [];
+        foreach ($this->node->getParamTagValues() as $paramTag) {
+            assert($paramTag instanceof ParamTagValueNode);
+            if (ltrim($paramTag->parameterName, '$') !== $paramName) {
+                continue;
+            }
+            return $this->convertType($paramTag->type);
+        }
+
+        return TypeFactory::undefined();
+    }
+
+    public function propertyType(string $propertyName): Type
+    {
+        $types = [];
+        foreach ($this->node->getPropertyTagValues() as $propertyTag) {
+            if (ltrim($propertyTag->propertyName, '$') !== $propertyName) {
+                continue;
+            }
+            return $this->convertType($propertyTag->type);
+        }
+
+        return TypeFactory::undefined();
+    }
+
+    public function formatted(): string
+    {
+        return implode(
+            "\n",
+            array_map(
+                static fn (string $line) => preg_replace('{^\s+}', '', $line),
+                explode(
+                    "\n",
+                    $this->node->__toString()
+                )
+            )
+        );
+    }
+
+    public function returnType(): Type
+    {
+        foreach ($this->node->getReturnTagValues() as $tag) {
+            return $this->convertType($tag->type);
+        }
+
+        foreach ($this->node->getReturnTagValues('@psalm-return') as $tag) {
+            return $this->convertType($tag->type);
+        }
+
+        return TypeFactory::undefined();
+    }
+
+    public function raw(): string
+    {
+        return $this->raw;
+    }
+
+    public function isDefined(): bool
+    {
+        return true;
+    }
+
+    public function properties(ReflectionClassLike $declaringClass): CoreReflectionPropertyCollection
+    {
+        $properties = [];
+        // merge read/write virtual properties
+        foreach ($this->node->getPropertyTagValues() as $propertyTag) {
```

Comment:
Here only `@property` tag is listed, without `@property-read` and `@property-write`, but I'm not sure if Phpactor has support for those

Reply:
Phpactor supports them AFAIK but only in the sense that it interprets them as `@property`

Reply:
Added `@property-read` and `@property-write` to parsed property types